code/game/gamemodes/objective.dm

```diff
@@ -580,16 +580,14 @@ GLOBAL_LIST_EMPTY(possible_items)
 	if(!dupe_search_range)
 		dupe_search_range = get_owners()
 	var/approved_targets = list()
-	check_items:
-		for(var/datum/objective_item/possible_item in GLOB.possible_items)
-			if(possible_item.objective_type != OBJECTIVE_ITEM_TYPE_NORMAL)
-				continue
-			if(!is_unique_objective(possible_item.targetitem,dupe_search_range))
-				continue
-			for(var/datum/mind/M in owners)
-				if(M.current.mind.assigned_role.title in possible_item.excludefromjob)
-					continue check_items
-			approved_targets += possible_item
+	for(var/datum/objective_item/possible_item in GLOB.possible_items)
+		if(!possible_item.valid_objective_for(owners))
+			continue
+		if(possible_item.objective_type != OBJECTIVE_ITEM_TYPE_NORMAL)
+			continue
+		if(!is_unique_objective(possible_item.targetitem,dupe_search_range))
+			continue
+		approved_targets += possible_item
```

Comment:
do note that I think this could be seriously problematic on lowpop for non-traitors since their steal objectives are permanent. I'm also introducing a change that gives traitors back their normal steal objectives so this could possibly lead to steal objectives actually being free objectives, which I don't think should be the case.

Reply:
I'd rather that permanent objectives have steal objectives for jobs that aren't flagged up right now rather than them having a free objective. That's what I mean.
Target-specific filters should only apply to temporary objectives that traitors are given.

Reply:
I understand the reasoning but I would _prefer_ to apply it to all antagonists anyway if possible.
My reasoning is that there are still several objectives in the pool which _don't_ have an owner requirement, specifically:

- Vault Documents
- The Nuclear Bomb Core
- Supermatter Sliver
- The Blackbox
- Functioning AI (this handily turns itself into a murder objective if there is no AI)

These are all pretty fun "heist" style things in my opinion, are unlikely to be invalidated when joining late, and will be valid to appear on any round.

Would it be acceptable as an alternative that I just remove the Captain/HoS presence requirement from some of their items so that they're always available and leave it such that it still doesn't generate CMO/RD/CE theft objectives if there isn't one?

The Captain's Laser especially is probably still fine with no Captain as its location makes it a fun heist and then after stealing it the temptation to use it is very strong.
The HoS laser too is behind the entire security wing and while that's often completely empty on lowpop it gives a sufficient obstacle to get in and similarly then leaves you with the temptation to out yourself by carrying it around.

Reply:
We discussed this on Discord and I'll just leave it only for secondary objectives.